Sources/ProjectDescription/Environment.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+@dynamicMemberLookup
+public struct Environment {
+    public enum Value {
+        case boolean(Bool)
+        case string(String)
+    }
+
+    public static subscript(dynamicMember member: String) -> Value? {
+        let snakeCaseMember = "TUIST_\(member.camelCaseToSnakeCase().uppercased())"
+        guard let value = ProcessInfo.processInfo.environment[snakeCaseMember] else { return nil }
+        let trueValues = ["1", "true", "TRUE", "yes", "YES"]
+        let falseValues = ["0", "false", "FALSE", "no", "NO"]
+        if trueValues.contains(value) {
+            return .boolean(true)
+        } else if falseValues.contains(value) {
+            return .boolean(false)
+        } else {
+            return .string(value)
+        }
+    }
+}
+
+private extension String {
+    func camelCaseToSnakeCase() -> String {
+        let acronymPattern = "([A-Z]+)([A-Z][a-z]|[0-9])"
```

Comment:
Regex is probably a fine solution as we want to allow only a limited amount of characters, but if we wanted something more robust, we can copy `JSONEncoder.convertFromSnakeCase` from [here](https://github.com/apple/swift/blob/88b093e9d77d6201935a2c2fb13f27d961836777/stdlib/public/Darwin/Foundation/JSONEncoder.swift#L161)

Reply:
I like the idea 😛. I've taken the implementation from there.